Build broken on newer libstdc++

**Describe the bug**

`rkmh.cpp` uses `std::numeric_limits`, which is in the `<limits>` standard library system header. On some limited, older versions of the C++ standard library, this was transitively included from `<algorithm>`, but it no longer is with newer C++ distributions.

This entire class of problems can be avoided with a tool like https://github.com/include-what-you-use/include-what-you-use . It's never safe to depend on transitive includes.

**To Reproduce**

Build on gcc 13 or clang 14. I don't know where the exact cutoff is, but you will _absolutely_ encounter this at some point when you upgrade to building on a newer OS or newer compilers.

```
/Users/alex/src/vcflib/src/rkmh.cpp: In function 'void rkmh::calc_hashes_(const char*, const uint64_t&, const uint64_t&, hash_t*&, int)':
/Users/alex/src/vcflib/src/rkmh.cpp:56:30: error: 'numeric_limits' is not a member of 'std'
   56 |             hashes[i] = std::numeric_limits<hash_t>::max();
      |                              ^~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
```



**Expected behavior**

vcflib should build with any c++17 compliant compiler.

--- a/src/rkmh.hpp
+++ b/src/rkmh.hpp
@@ -7,6 +7,7 @@
 #include <unordered_set>
 #include <math.h>
 #include <algorithm>
+#include <limits>
 #include "murmur3.hpp"
 
 // From Eric's https://github.com/edawson/rkmh

